### PR TITLE
Reorganize repositories

### PR DESCRIPTION
**Pull request by aherbst-broad**
## Why

We were including jobs in the prod repo that didn't belong there.

## This PR
* Reorganize jobs in each repo
* Fixes a few bugs

## Checklist
- [ ] Documentation has been updated as needed.

### CODE REVIEW

File: orchestration/hca_manage/snapshot.py [hunk 1]
```diff
@@ -208,8 +208,6 @@ def __post_init__(self) -> None:
             "dev": [
                 "hca-snapshot-readers@dev.test.firecloud.org",
                 "monster-dev@dev.test.firecloud.org",
-                "azul-dev@dev.test.firecloud.org",
```
Comment (aherbst-broad) [inline orchestration/hca_manage/snapshot.py:211]: We don't need to add these groups anymore as snapshots are public by default.

File: orchestration/hca_orchestration/repositories/base_repositories.py [hunk 2]
```diff
@@ -36,14 +36,11 @@ def copy_project_to_new_dataset_job() -> PipelineDefinition:
         })
 
 
-def base_jobs() -> list[Union[PipelineDefinition, SensorDefinition]]:
+def base_jobs(mode: str) -> list[Union[PipelineDefinition, SensorDefinition]]:
     defs = [
         cut_snapshot,
         load_hca,
-        validate_egress,
-        build_post_import_sensor(os.environ.get("ENV", "test")),
```
Comment (aherbst-broad) [inline orchestration/hca_orchestration/repositories/base_repositories.py:44]: build_post_import_sensor is not used anymore (it was triggered when our old argo stuff ran).
copy_project_to_new_dataset_job only makes sense in a dev context.

File: orchestration/hca_orchestration/repositories/base_repositories.py [hunk 3]
```diff
@@ -36,14 +36,11 @@ def copy_project_to_new_dataset_job() -> PipelineDefinition:
...
-        copy_project_to_new_dataset_job(),
+        validate_egress
     ]
 
-    return defs + load_dcp_release_manifests() + copy_project_to_new_dataset_partitions() + \
-        dev_refresh_cut_snapshot_partition_set()
```
Comment (aherbst-broad) [inline orchestration/hca_orchestration/repositories/base_repositories.py:49]: Same comments here, copy_project_to_new_dataset_partitions and dev_refresh_cut_snapshot_partition_set only make sense in dev.

File: orchestration/hca_orchestration/solids/create_snapshot.py [hunk 4]
```diff
@@ -75,7 +75,7 @@ def add_steward(context: AbstractComputeExecutionContext, snapshot_id: str) -> s
         if policy.name == 'steward':
             found_member = False
             for member in policy.members:
-                if policy_member == member.email:
+                if policy_member == member:
```
Comment (aherbst-broad) [inline orchestration/hca_orchestration/solids/create_snapshot.py:78]: This was a bug I bumped into in dev.

**Review by quazi-broad — APPROVED** [review]
Makes sense to move those dev pipelines out of base_repositories.

**Pull request merged**